openwisp_monitoring/device/api/views.py

```diff
@@ -200,7 +202,29 @@ def _write(self, request, pk):
                 metric.write(client['mac'])
             if created:
                 self._create_clients_graph(metric)
+        for resource, value in data.get('resources', {}).items():
```

Comment:
This for loop looks weird.
Instead of iterating over data.get('resources') IMHO it makes more sense to directly iterate over self.resources like

```
for resource in self.resources:
    value = data.get('resources', {}).get(resource)
```

or directly (even better):

```
if 'memory' in data.get('resources', {}):
   ...
if 'load' in data.get('resources', {}):
   ...
if 'disk_usage' in data.get('resources', {}):
   ...
```

Why are you using data.get('resources') instead of data['resources']? Is there any case when resources will not be available?

Reply:
I used it for two reason one was to maintain consistency and the other was to avoid garbage clients or blank data from failing the API but better would be to use an `if` it doesn't make much sense to use a `for` as I had done so updated that :smile: